### ISSUE TITLE
Support for exceptions

### ISSUE THREAD

**Issue by bismark** [peek-travel/excal#40]
Would there be any interest in adding support for recurrence exceptions? 

I've had to extend this lib to support exceptions for our needs so I wanted to open the discussion about pushing this functionality upstream.

In particular I've extended the NIF to support parsing `EXDATE` strings to return a list of date (times) and I've built a new `Stream.resource` that takes in the RRULE and a set of exception date(times) to generate the correct set of occurrences.

**Comment by doughsay** [comment]
Yeah, I'm very interested in supporting this. I wasn't sure how I was going to go about it, but it sounds like your approach is good. The idea is to eventually support entire ical event components, which could consist of an RRULE with many EXDATE lines in it, so this is definitely within the realm of what this library is meant to evolve into.
